examples/medical_image_segmentation.ipynb

```diff
@@ -577,11 +578,12 @@
     "        _, targets = batch\n",
     "        return self.criterion(outputs, targets)\n",
     "\n",
-    "    def get_metrics(self, is_train: bool = False):\n",
-    "        if is_train:\n",
-    "            return {'BCEWithLogitsLoss': self.train_loss}\n",
-    "        else:\n",
-    "            return {'BCEWithLogitsLoss': self.val_loss, 'Dice': self.dice}"
+    "    def metrics(self, train: bool = False):\n",
```

Comment:
When I go back to the previous implementation I get `AttributeError: 'SMPUNet' object has no attribute 'dice'`. I assume the reason I wasn't getting this error is somewhere in the difference between `def metrics ...` and `def get_metrics ...`

Should `return {'BCEWithLogitsLoss': self.val_loss, 'Dice': self.dice}` actually be `return {'BCEWithLogitsLoss': self.val_loss, 'Dice': self.val_dice}` ?

Because I can see this definition `self.val_dice = Dice(num_classes=classes)` 

If I do change it to `... 'Dice': self.val_dice` I still see get issues but if you think it should stay as `self.dice` I will hold off on resolving those issues.

Reply:
It does look like it should be `self.val_dice` :) I would go down that route. `get_metrics` is definitely the right function name, and returning a `Dict[str, Metrci]` is also correct.